src/nvim/ops.c

```diff
@@ -1709,7 +1709,14 @@ static void mb_adjust_opend(oparg_T *oap)
 static inline void pbyte(pos_T lp, int c)
 {
   assert(c <= UCHAR_MAX);
-  *(ml_get_buf(curbuf, lp.lnum, true) + lp.col) = (char_u)c;
+
+  char_u * line = ml_get_buf(curbuf, lp.lnum, true);
+  ssize_t old_len = (ssize_t)STRLEN(line);
+
+  line[lp.col] = (char_u)c;
+
+  ml_add_deleted_len(line, old_len);
```

Comment:
the true in `ml_get_buf(curbuf, lp.lnum, true)` sets the line dirty so I suppose it should update statistics at some point. Maybe it happens too late ?

Reply:
After some digging, it seems (I may be wrong here) that the `buf->deleted_bytes` is not related to the fact that the line is dirty.

Reply:
Plus the name `deleted_bytes` is actually misleading, as this represent the old byte size for buffer updates.
I think that this `old_byte_size` should be computed just before sending the buffer update, in the extmarks code.

Reply:
since old_byte_size has to be lines anyway, shouldn't it be updated here https://github.com/neovim/neovim/blob/master/src/nvim/memline.c#L1879 instead ? 
There are some other calls of `ml_get_buf(_, _, true)` that won't be taken care of otherwise (I could find examples throughout the code. This can be revisited in the bytetrack PR (maybe add it as a comment so that we don't forget).

Reply:
Yep that's far better